grails-app/services/au/org/ala/phyloviz/TreeService.groovy

```diff
@@ -670,4 +679,113 @@ class TreeService {
 
         Tree.findById(treeId)?.delete()
     }
-}
+
+    /**
+     * Trim the specified tree using the provided option.
+     *
+     * @param treeId Id of the tree to trim. The tree can be either newick or nexml
+     * @param option The {@link TrimOption} to use
+     * @param trimToInclude True to INCLUDE the identified species, false to EXCLUDE the identified species
+     * @param data Additional data required by the trim option (e.g. species list DataResourceId if TrimOption#SPECIES_LIST is selected)
+     * @return The trimmed tree in newick format.
+     */
+    Tree trimTree(Integer treeId, TrimOption option, boolean trimToInclude, data = null) {
+        log.debug("Before : ${getSpeciesNamesFromTree(treeId).size()}")
+
+        Tree tree = Tree.findById(treeId)
```

Comment:
can move this after TrimOption.None statement. otherwise, you are simply querying the db.